lib/bank_accounts/checking.rb

```diff
@@ -0,0 +1,55 @@
+require 'pry'
+
+module BankAccounts
+  class CheckingAccount < Account
+
+    attr_reader :balance, :checks_used
+
+    def initialize(id, initial_balance)
+      # inherit from Account class
+      super
+
+      # start with 0 checks used
+      @checks_used = 0
+    end
+
+    def withdraw(amount)
+      #incur a fee of $1 with each withdrawal
+      amount += 1
+      super
+      return round_to_hundredths(@balance)
+    end
+
+    def withdraw_using_check(amount)
+      @checks_used += 1 # increase checks used with each method call
+
+      if @checks_used > 3
+        # $2 fee for greater than 3 checks
+        amount += 2
+
+        # Raises warning message if overdraft goes lower than -$10
+        if (@balance - amount) < -10
+          raise ArgumentError.new "OVERDRAFT WARNING - only $10 overdraft allowed. Your current balance is #{round_to_hundredths(@balance)}."
+        else
+          @balance -= amount
+        end
+        return @balance = round_to_hundredths(@balance)
+
+      # no transaction fee for 3 or fewer checks
+      else
+        if (@balance - amount) < -10
+          raise ArgumentError.new "OVERDRAFT WARNING - only $10 overdraft allowed. Your current balance is #{round_to_hundredths(@balance)}."
+        else
+          @balance -= amount
+        end
+        return @balance = round_to_hundredths(@balance)
+      end
+    end
```

Comment:
Great job getting this working! That is the most important thing! 

You may notice that you have a lot of duplication in structure here. One option for refactor might be moving around the order of the control flow. Lines 31-36 are identical to lines 40-45. 

When I see these nestings I reach for a couple tools. 

One is reorganization. Here I'm thinking, do I really care about the number of checks used if they are overdrawn and we won't cash their check anyway? Perhaps we should check for that first and only deal with checks if we would process them. 

The other tool I reach for is abstraction. Can I pull out any chunks of code to make their own method to make this easier to read. One thing to note here is that in the regular withdrawal method we call super to reuse code that is already written in the parent class. Can we do that anywhere in the withdraw_using_check class?